### PR TITLE
Added NotContainMatch method for string collections and unit tests to cover it

### PR DESCRIPTION
**Pull request by tomaustin700**
Added NotContainMatch for string collections which does the opposite of ContainMatch. I recently had a need for NotContainMatch so decided to add it.

### CODE REVIEW

File: FluentAssertions.sln [hunk 1]
```diff
@@ -48,12 +48,11 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NetCore30.Specs", "Tests\Ne
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Approval.Tests", "Tests\Approval.Tests\Approval.Tests.csproj", "{F5115158-A038-4D14-A04E-46E7863E40B9}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "NetCore21.Specs", "Tests\NetCore21.Specs\NetCore21.Specs.csproj", "{0CF100F2-CE71-41F4-9523-62AD78BD6961}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "NetCore21.Specs", "Tests\NetCore21.Specs\NetCore21.Specs.csproj", "{0CF100F2-CE71-41F4-9523-62AD78BD6961}"
```
Comment (dennisdoomen) [inline FluentAssertions.sln]: What happened here?

File: Src/FluentAssertions/Collections/StringCollectionAssertions.cs [hunk 2]
```diff
@@ -242,5 +242,58 @@ private IEnumerable<string> AllThatMatch(string wildcardPattern)
...
+        public AndWhichConstraint<StringCollectionAssertions, string> NotContainMatch(string wildcardPattern, string because = "",
+            params object[] becauseArgs)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(Subject is object)
+                .FailWith("Expected {context:collection} to not contain a match of {0}{reason}, but found <null>.", wildcardPattern)
+                .Then
+                .ForCondition(NotContainsMatch(wildcardPattern))
+                .FailWith("Expected {context:collection} {0} to not contain a match of {1}{reason}.", Subject, wildcardPattern);
```
Comment (dennisdoomen) [inline Src/FluentAssertions/Collections/StringCollectionAssertions.cs]: I would probably rephrase this as `Did not expect`

File: Src/FluentAssertions/Collections/StringCollectionAssertions.cs [hunk 3]
```diff
@@ -242,5 +242,58 @@ private IEnumerable<string> AllThatMatch(string wildcardPattern)
...
+            {
+                return Subject.All(item =>
+                {
+                    item.Should().NotMatch(wildcardPattern);
+                    return !scope.Discard().Any();
+                });
+            }
+        }
+
+        private IEnumerable<string> AllThatDontMatch(string wildcardPattern)
```
Comment (jnyrup) [inline Src/FluentAssertions/Collections/StringCollectionAssertions.cs]: Avoid contractions.
`AllThatDontMatch` -> `AllThatDoNotMatch`

File: FluentAssertions.sln (hunk 1 above)
Comment (tomaustin700) [reply FluentAssertions.sln]: Not sure what happened, should now be back to normal.

File: Src/FluentAssertions/Collections/StringCollectionAssertions.cs (hunk 2 above)
Comment (tomaustin700) [reply Src/FluentAssertions/Collections/StringCollectionAssertions.cs]: Changed to "Did not expect"

File: Src/FluentAssertions/Collections/StringCollectionAssertions.cs (hunk 3 above)
Comment (tomaustin700) [reply Src/FluentAssertions/Collections/StringCollectionAssertions.cs]: Changed to `AllThatDoNotMatch`

**Comment by tomaustin700** [comment]
@dennisdoomen @jnyrup Thanks for the review, all issues should hopefully now be resolved.

File: Src/FluentAssertions/Collections/StringCollectionAssertions.cs [hunk 4]
```diff
@@ -242,5 +242,58 @@ private IEnumerable<string> AllThatMatch(string wildcardPattern)
...
+        /// The wildcard pattern with which the subject is matched, where * and ? have special meanings.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndWhichConstraint<StringCollectionAssertions, string> NotContainMatch(string wildcardPattern, string because = "",
```
Comment (jnyrup) [inline Src/FluentAssertions/Collections/StringCollectionAssertions.cs]: Wondering whether a negative assertion should return an `AndWhichConstraint`...
I have an odd feeling about asserting that a collection does not contains something and then continue asserting on what *didn't* match.
Also because `.Which` will only(?) make sense if the collection has *exactly* one non-matching item, which can be returned from `.Which`.
Maybe I'm just struggling finding a good example.

Comment (dennisdoomen) [reply Src/FluentAssertions/Collections/StringCollectionAssertions.cs]: No, I think you are right. Let's just use an `AndConstraint`

Comment (tomaustin700) [reply Src/FluentAssertions/Collections/StringCollectionAssertions.cs]: Just pushed this change :)
